code/parachain/frame/staking-rewards/src/lib.rs

```diff
@@ -1597,22 +1584,25 @@ fn add_to_rewards_pot<T: Config>(
 	amount: T::Balance,
 	keep_alive: bool,
 ) -> DispatchResult {
-	RewardPools::<T>::get(pool_id)
-		.ok_or(Error::<T>::RewardsPoolNotFound)?
-		.rewards
-		.get(&asset_id)
-		.ok_or(Error::<T>::RewardAssetNotFound)?;
+	RewardPools::<T>::try_mutate(pool_id, |pool| {
+		let pool = pool.as_mut().ok_or(Error::<T>::RewardsPoolNotFound)?;
 
-	let pool_account = Pallet::<T>::pool_account_id(&pool_id);
+		let reward = pool.rewards.get_mut(&asset_id).ok_or(Error::<T>::RewardAssetNotFound)?;
 
-	T::Assets::transfer(asset_id, &who, &pool_account, amount, keep_alive)?;
-	T::Assets::hold(asset_id, &pool_account, amount)?;
+		if RewardsPotIsEmpty::<T>::contains_key(pool_id, asset_id) {
+			reward.last_updated_timestamp = T::UnixTime::now().as_secs();
```

Comment:
why are you updating last_updated_timestsamp here, when you're not accumulating any rewards?

Reply:
This is to avoid retroactively accumulating rewards. When it accumulates the next time, it will start from this timestamp instead of the last time it accumulated. Originally I added a `last_time_pot_funded: Option<u64>` field which served this purpose. Do you think that's a better approach?

Reply:
Ah I see now, I think that makes more sense than updating every block. We could even change the `RewardPotIsEmpty` storage to contain the last updated timestamp as an `Option<u64>` (since we have to read it anyways), with `None` meaning the pot is empty.

We could also consider a more descriptive enum, something like:

```rust
enum RewardsPotState {
    Paused,
    Running {
        last_updated_timestamp: u64
    },
}
```

The `last_updated_timestamp` field could then be removed from the pool storage, which would make more sense.

Reply:
Yeah that could work as well, but that would mean reading the `RewardPotIsEmpty` storage every block to calculate the accumulated rewards. I'm not sure how often will pools be paused (in which case we avoid getting the whole pool from storage) and how often now (which will mean having an extra read compared to what we have now)

Reply:
this pallet isn't live, so we can change the storage